### ISSUE TITLE
undefined method `path' for class Selenium::WebDriver::DriverFinder (NoMethodError) with selenium-webdriver 4.20.0 

### ISSUE THREAD

**Issue by yahonda** [rails/rails#51658]
Managed to reproduce CI failure at https://buildkite.com/rails/rails-nightly/builds/457#018f128f-00fe-4598-b07a-73c67fb56e44/1136-1138 .

### Steps to reproduce
```ruby
git clone https://github.com/rails/rails
cd rails
bundle update selenium-webdriver --conservative
cd actionpack
git diff # to show the selenium-webdriver version
bin/test
```

### Expected behavior
It should pass.

### Actual behavior
It raises "undefined method `path' for class Selenium::WebDriver::DriverFinder (NoMethodError)".

```diff
$ git diff
diff --git a/Gemfile.lock b/Gemfile.lock
index 82be41de68..081a26d484 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -491,7 +491,8 @@ GEM
       google-protobuf (~> 3.25)
     sass-embedded (1.69.6-x86_64-linux-gnu)
       google-protobuf (~> 3.25)
-    selenium-webdriver (4.16.0)
+    selenium-webdriver (4.20.0)
+      base64 (~> 0.2)
       rexml (~> 3.2, >= 3.2.5)
       rubyzip (>= 1.2.2, < 3.0)
       websocket (~> 1.0)
$
```

```ruby
$ bin/test
/home/yahonda/src/github.com/rails/rails/actionpack/lib/action_dispatch/system_testing/browser.rb:75:in `resolve_driver_path': undefined method `path' for class Selenium::WebDriver::DriverFinder (NoMethodError)

          namespace::Service.driver_path = ::Selenium::WebDriver::DriverFinder.path(options, namespace::Service)
                                                                              ^^^^^
	from /home/yahonda/src/github.com/rails/rails/actionpack/lib/action_dispatch/system_testing/browser.rb:45:in `preload'
	from /home/yahonda/src/github.com/rails/rails/actionpack/lib/action_dispatch/system_testing/driver.rb:21:in `initialize'
	from /home/yahonda/src/github.com/rails/rails/actionpack/lib/action_dispatch/system_test_case.rb:161:in `new'
	from /home/yahonda/src/github.com/rails/rails/actionpack/lib/action_dispatch/system_test_case.rb:161:in `driven_by'
	from /home/yahonda/src/github.com/rails/rails/actionpack/test/abstract_unit.rb:519:in `<class:DrivenBySeleniumWithChrome>'
	from /home/yahonda/src/github.com/rails/rails/actionpack/test/abstract_unit.rb:518:in `<top (required)>'
	from /home/yahonda/.rbenv/versions/3.3.1/lib/ruby/3.3.0/bundled_gems.rb:74:in `require'
	from /home/yahonda/.rbenv/versions/3.3.1/lib/ruby/3.3.0/bundled_gems.rb:74:in `block (2 levels) in replace_require'
	from /home/yahonda/src/github.com/rails/rails/actionpack/test/abstract/callbacks_test.rb:3:in `<top (required)>'
	from /home/yahonda/.rbenv/versions/3.3.1/lib/ruby/3.3.0/bundled_gems.rb:74:in `require'
	from /home/yahonda/.rbenv/versions/3.3.1/lib/ruby/3.3.0/bundled_gems.rb:74:in `block (2 levels) in replace_require'
	from /home/yahonda/src/github.com/rails/rails/railties/lib/rails/test_unit/runner.rb:61:in `block in load_tests'
	from /home/yahonda/src/github.com/rails/rails/railties/lib/rails/test_unit/runner.rb:60:in `each'
	from /home/yahonda/src/github.com/rails/rails/railties/lib/rails/test_unit/runner.rb:60:in `load_tests'
	from /home/yahonda/src/github.com/rails/rails/railties/lib/rails/test_unit/runner.rb:52:in `run'
	from /home/yahonda/src/github.com/rails/rails/tools/test.rb:18:in `<top (required)>'
	from bin/test:5:in `require_relative'
	from bin/test:5:in `<main>'
$
```

### System configuration
**Rails version**: main branch

**Ruby version**: ruby 3.3.1 (2024-04-23 revision c56cd86388) [x86_64-linux]

**Comment by Earlopain** [comment]
I asked if this is an intentional change since the PR description made it sound like it should be deprecated instead: https://github.com/SeleniumHQ/selenium/pull/13386#issuecomment-2076465418

**Comment by yahonda** [comment]
Let's see how https://github.com/SeleniumHQ/selenium/pull/13877 goes. In the meantime let's pin the selenium-webdriver version to 4.19.0 .

**Comment by yahonda** [comment]
Opened https://github.com/rails/rails/pull/51662 to pin the selenium-webdriver version .
Feel free to open a pull request to support the newer versions of selenium-webdriver that deprecates `Selenium::WebDriver::DriverFinder.path`.

**Comment by p0deje** [comment]
This was accidentally removed API that should have been deprecated instead. Sorry, should be fixed in 4.20.1.
